prawcore/auth.py

```diff
@@ -331,23 +331,27 @@ class ScriptAuthorizer(Authorizer):
 
     AUTHENTICATOR_CLASS = TrustedAuthenticator
 
-    def __init__(self, authenticator, username, password):
+    def __init__(self, authenticator, username, password, otp_function=None):
         """Represent a single personal-use authorization to Reddit's API.
 
         :param authenticator: An instance of :class:`TrustedAuthenticator`.
         :param username: The Reddit username of one of the application's
             developers.
         :param password: The password associated with ``username``.
+        :param otp_function: A function that returns OTPs (One-Time Passcodes),
+            also known as 2FA auth codes.
```

Comment:
```suggestion
            also known as 2FA auth codes. If this function is provided,
            prawcore will call it when authenticating, each time it needs
            an OTP code.
```

Reply:
You might want to consider rewording the added line. If the param is supplied, the callback is executed every time that prawcore requests a token.  Assuming that "authenticating" refers to every act of obtaining an access token,  and not just the initial one,  "when authenticating" and "each time it needs an OTP code" are the same thing.